### PR TITLE
BaSP-MR-62: Subscriptions screen

### PR DESCRIPTION
**Pull request by leo-balonchard**
I worked on the subscription resource together with @GinoLucci1  

Added:
- Table component
- Form component
- Add function
- Delete function
- Edit function

### CODE REVIEW

File: src/Components/Subscriptions/Table/index.jsx [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+              {expandedRows.includes(item._id) && (
+                <tr>
+                  <td colSpan="8">
+                    <select
+                      className={styles.selectEdit}
+                      name="editMember"
+                      value={user.idMember}
+                      onChange={onChangeInput}
+                    >
+                      <option value="646014b31c70e12b863ad70a">Juan Manuel Lopez</option>
```
Comment (CavalloFede) [inline src/Components/Subscriptions/Table/index.jsx:178]: same as before, we need to change those hardcode ids to ones that we are sure exist on the DB

File: src/Components/Subscriptions/Table/index.jsx [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+                <td>
+                  <button className={styles.button} onClick={() => handleEdit(item._id)}>
+                    Edit
+                  </button>
+                  <button className={styles.buttonDelete} onClick={() => handleDelete(item._id)}>
+                    Delete
+                  </button>
+                </td>
+              </tr>
+              {expandedRows.includes(item._id) && (
```
Comment (CavalloFede) [inline src/Components/Subscriptions/Table/index.jsx:169]: We are missing the class on the edit form

File: src/Components/Subscriptions/index.jsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  };
+
+  useEffect(() => {
+    fetch(`${process.env.REACT_APP_API_URL}/api/subscriptions/`)
+      .then((response) => response.json())
+      .then((data) => {
+        setSubscriptions(data);
+      })
+      .catch((error) => {
+        console.error('Error:', error);
```
Comment (CavalloFede) [inline src/Components/Subscriptions/index.jsx:25]: we can delete de console.log

File: src/Components/Subscriptions/Form/index.jsx [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    fetch(`${process.env.REACT_APP_API_URL}/api/subscriptions/`, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(requestData)
+    })
+      .then((response) => response.json())
+      .then((data) => {
+        console.log(data);
```
Comment (CavalloFede) [inline src/Components/Subscriptions/Form/index.jsx:36]: We can delete console.log

File: src/Components/Subscriptions/Table/index.jsx [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+    fetch(`${process.env.REACT_APP_API_URL}/api/subscriptions/${id}`, {
+      method: 'PUT',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(requestData)
+    })
+      .then((response) => response.json())
+      .then((data) => {
+        console.log(data);
```
Comment (CavalloFede) [inline src/Components/Subscriptions/Table/index.jsx:97]: We can delete console.log

File: src/Components/Subscriptions/Table/index.jsx [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+    fetch(`${process.env.REACT_APP_API_URL}/api/subscriptions/`, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(requestData)
+    })
+      .then((response) => response.json())
+      .then((data) => {
+        console.log(data);
```
Comment (CavalloFede) [inline src/Components/Subscriptions/Table/index.jsx:53]: We can delete console.log

File: src/Components/Subscriptions/Table/index.jsx [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+  };
+
+  const handleDelete = (id) => {
+    alert('Subscription ' + id + ' removed successfully');
+    fetch(`${process.env.REACT_APP_API_URL}/api/subscriptions/${id}`, {
+      method: 'DELETE'
+    })
+      .then((response) => response.json())
+      .then((data) => {
+        console.log(data);
```
Comment (CavalloFede) [inline src/Components/Subscriptions/Table/index.jsx:68]: We can delete console.log

**Review by CavalloFede — CHANGES_REQUESTED** [review]

**Review by CavalloFede — APPROVED** [review]

**Review by iaracriscenti — APPROVED** [review]
🚀

**Pull request merged**